Auth0/JWKS.swift

```diff
@@ -0,0 +1,91 @@
+// JWKS.swift
+//
+// Copyright (c) 2019 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import SimpleKeychain
+
+public struct JWKS: Codable {
+    let keys: [JWK]
+}
+
+public extension JWKS {
+    func key(id kid: String) -> JWK? {
+        return keys.first { $0.keyId == kid }
+    }
+}
+
+public struct JWK: Codable {
+    let keyType: String
+    let keyId: String?
+    let usage: String?
+    let algorithm: String?
+    let certUrl: String?
+    let certThumbprint: String?
+    let certChain: [String]?
+    let rsaModulus: String?
+    let rsaExponent: String?
+
+    enum CodingKeys: String, CodingKey {
+        case keyType = "kty"
+        case keyId = "kid"
+        case usage = "use"
+        case algorithm = "alg"
+        case certUrl = "x5u"
+        case certThumbprint = "x5t"
+        case certChain = "x5c"
+        case rsaModulus = "n"
+        case rsaExponent = "e"
+    }
+}
+
+extension JWK {
+    var rsaPublicKey: SecKey? {
+        guard let modulus = rsaModulus?.a0_decodeBase64URLSafe(),
+            let exponent = rsaExponent?.a0_decodeBase64URLSafe() else { return nil }
```

Comment:
would this "else return nil" apply to the modulus or only to the exponent as well?

Reply:
Would apply to both of them. It's equivalent to the following if:
```swift
    if rsaModulus?.a0_decodeBase64URLSafe() != nil && rsaExponent?.a0_decodeBase64URLSafe() != nil {}
```

Reply:
Would apply to both of them. It's equivalent to the following if:
```swift
    if rsaModulus?.a0_decodeBase64URLSafe() != nil && rsaExponent?.a0_decodeBase64URLSafe() != nil {
        let modulus = rsaModulus!.a0_decodeBase64URLSafe()!
        let exponent = rsaExponent!.a0_decodeBase64URLSafe()!
}

```